ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JClient.java

```diff
@@ -88,23 +93,33 @@ private boolean isAuthenticationException(final Throwable exception) {
     return message.contains("received: 401") || message.contains("received: 403");
   }
 
+  private boolean isTimeoutException(final Throwable exception) {
+    return exception instanceof TimeoutException;
+  }
+
   protected void handleError(final Throwable error) {
     handleError(error, false);
   }
 
   protected void handleError(final Throwable error, final boolean couldBeAuthError) {
-    final long errorTime = lastError.get();
-    if (errorTime == NO_ERROR_TIME
-        || timeProvider.getTimeInMillis().longValue() - errorTime > ERROR_REPEAT_DELAY_MILLIS) {
-      if (lastError.compareAndSet(errorTime, timeProvider.getTimeInMillis().longValue())) {
-        eventLog.executionClientIsOffline(error, couldBeAuthError);
+    final long lastErrorTime = lastError.get();
+    final long timeNow = timeProvider.getTimeInMillis().longValue();
```

Comment:
its isn't important but we may want to update lastErrorTime atomically (maybe using `nowOrLastErrorTime = accumulateAndGet(timeNow, (,)) - > {...}`)